### PR TITLE
Add --ini-file cmdline option

### PR DESCRIPTION
**Pull request by Helldiver-RX**
Makes it possible to change the name of the ini configuration file. The default name is still "diablo.ini".

### CODE REVIEW

**Comment by StephenCWills** [comment]
Is there an advantage to using this over `--config-dir`?

File: Source/diablo.cpp [hunk 1]
```diff
@@ -995,6 +996,12 @@ void DiabloParseFlags(int argc, char **argv)
 				diablo_quit(64);
 			}
 			paths::SetConfigPath(argv[++i]);
+    } else if (arg == "--ini-file") {
+			if (i + 1 == argc) {
+				PrintFlagsRequiresArgument("--ini-file");
+				diablo_quit(64);
+			}
+			paths::SetIniFile(argv[++i]);
```
Comment (qndel) [inline Source/diablo.cpp:1004]: garbage formatting

**Comment by StephenCWills** [comment]
> It is just an extra cmdline option and does something different: --config-dir sets the directory where "diablo.ini" is and --ini-file sets an optional new name for the file.

I'm not so much asking about what it does as opposed to what it's used for. In other words, I get that it changes the name of the file, but why do you want to change the name of the file?

**Comment by DakkJaniels** [comment]
I forget, does `--config-dir` use the save games in that directory? If so, I guess that could be one possible use case (use different configuration, but still use same save games). Not sure how often that's really needed though.

**Comment by Helldiver-RX** [comment]
> > It is just an extra cmdline option and does something different: --config-dir sets the directory where "diablo.ini" is and --ini-file sets an optional new name for the file.
> 
> I'm not so much asking about what it does as opposed to what it's used for. In other words, I get that it changes the name of the file, but why do you want to change the name of the file?

One possible use would be to keep several ini files in the same folder, each one with a different set of configuration values for the program.

**Comment by Helldiver-RX** [comment]
> I forget, does `--config-dir` use the save games in that directory? If so, I guess that could be one possible use case (use different configuration, but still use same save games). Not sure how often that's really needed though.

Yes, that is what i had in mind with this.

**Comment by StephenCWills** [comment]
> One possible use would be to keep several ini files in the same folder, each one with a different set of configuration values for the program.

Is this how you intend to use it? And is there an advantage to this over having multiple config directories in the same folder?

**Comment by FitzRoyX** [comment]
> One possible use would be to keep several ini files in the same folder, each one with a different set of configuration values for the program.

You can already manually swap files out. Why would you need to flip between configs quickly like that? If this was added, you'd be the only person using it and mainline features should be something that at least a third of users will use.

**Pull request closed without merging**